rubi/tests/rubi_tests.py

```diff
@@ -0,0 +1,889 @@
+        assert client.wallet == account_1["address"]
+        # Test if the key attribute is set correctly when a key is provided.
+        assert client.key == account_1['key']
+        # Test if the market/router have correct types and are init
+        assert isinstance(client.market, RubiconMarket)
+        assert isinstance(client.router, RubiconRouter)
+        # Test if the _pairs attribute is initialized as an empty dictionary.
+        assert len(client._pairs.keys()) == 0
+        # Test if the message_queue attribute is set to None when no queue is provided.
+        assert client.message_queue is None
+
+    ######################################################################
+    # pair tests
+    ######################################################################
+
+    def test_add_pair(
+        self,
+        test_client: Client,
+        cow: Contract,
+        eth: Contract
+    ):
+        pair_name = "COW/ETH"
+
+        test_client.add_pair(pair_name=pair_name)
+
+        assert len(test_client.get_pairs_list()) == 1
+        assert test_client.get_pairs_list()[0] == pair_name
+
+        pair = test_client.get_pair(pair_name=pair_name)
+        assert pair.base_asset.address == cow.address
+        assert pair.quote_asset.address == eth.address
+
+    def test_update_pair_allowance(
+        self,
+        rubicon_market: Contract,
+        test_client_for_account_1: Client,
+        cow_erc20_for_account_1: ERC20,
+        eth_erc20_for_account_1: ERC20
+    ):
+        pair_name = "COW/ETH"
+
+        initial_cow_allowance = cow_erc20_for_account_1.allowance(
+            owner=cow_erc20_for_account_1.wallet,
+            spender=rubicon_market.address
+        )
+        initial_eth_allowance = eth_erc20_for_account_1.allowance(
+            owner=eth_erc20_for_account_1.wallet,
+            spender=rubicon_market.address
+        )
+
+        # in fixtures these are initialized with the max approval value
+        max_approval = 2 ** 256 - 1
+
+        assert initial_cow_allowance == max_approval
+        assert initial_eth_allowance == max_approval
+
+        test_client_for_account_1.update_pair_allowance(
+            pair_name=pair_name,
+            new_base_asset_allowance=Decimal("12"),
+            new_quote_asset_allowance=Decimal("100"),
+        )
+
+        new_cow_allowance = cow_erc20_for_account_1.allowance(
+            owner=cow_erc20_for_account_1.wallet,
+            spender=rubicon_market.address
+        )
+        new_eth_allowance = eth_erc20_for_account_1.allowance(
+            owner=eth_erc20_for_account_1.wallet,
+            spender=rubicon_market.address
+        )
+
+        assert new_cow_allowance != initial_cow_allowance
+        assert new_cow_allowance == cow_erc20_for_account_1.to_integer(Decimal("12"))
+
+        assert new_eth_allowance != initial_eth_allowance
+        assert new_eth_allowance == eth_erc20_for_account_1.to_integer(Decimal("100"))
+
+    def test_delete_pair(
```

Comment:
may be overly verbose, but also checking that token allowances are set to zero here would be nice